City-of-Helsinki/drupal-hdbt
# UHF-1774 Block Revisions

Prepare the site: 
1. `composer create-project City-of-Helsinki/drupal-helfi-platform:dev-main hel-platform --no-interaction --repository https://repository.drupal.hel.ninja/`
2. `cd hel-platform`
3. `composer require drupal/hdbt:dev-UHF-1774_block_revisions && composer require drupal/hdbt_admin:dev-UHF-1774_block_revisions && composer require drupal/helfi_tpr:dev-UHF-1774_block_revisions`
4. `make new`
5. Login to the site.
6. `make shell`
7. `drush en helfi_tpr_config -y`
8. `MIGRATE_LIMIT=5 drush mim tpr_unit && MIGRATE_LIMIT=5 drush mim tpr_service && drush cr`

How to test:
1. Go to create a basic page 
2. Add hero, sidebar text and some paragraph in to lower content
3. Save the page and edit the page again
4. Change something in hero and sidebar and maybe add something new to the page
5. Save the page and go to revisions
6. Check that the earlier revision exists and differs from newest version (don't just compare the pages, open the revisions in different tabs) ❗  
7. Create a translation of the page and repeat steps 2..6
8. Once satisfied with basic page, repeat the same procedures to Landing page, Unit and Service. _Notice. Landing page and Unit doesn't have sidebar and/or lower content_ 

Other updates in this PR:
- Node version for build tools
- Recommended node modules (by renovate)

Check also these:
- https://github.com/City-of-Helsinki/drupal-hdbt-admin/pull/66
- https://github.com/City-of-Helsinki/drupal-module-helfi-tpr/pull/62
